pkg/server/infrastructure/datastore/mysql/mysql.go

```diff
@@ -47,40 +64,249 @@ func New(ctx context.Context, cfg datastore.Config) (datastore.DataStore, error)
 
 // Add add data model
 func (m *mysql) Add(ctx context.Context, entity datastore.Entity) error {
-	return fmt.Errorf("method to be implemented")
+	if entity.PrimaryKey() == "" {
+		return datastore.ErrPrimaryEmpty
+	}
+	if entity.TableName() == "" {
+		return datastore.ErrTableNameEmpty
+	}
+	entity.SetCreateTime(time.Now())
+	entity.SetUpdateTime(time.Now())
+
+	if dbAdd := m.client.Create(entity); dbAdd.Error != nil {
```

Comment:
I'll comment here but this should to be done for all operations.

Operations should use `ctx`. use client like

```go
func (m *mysql) client(ctx context.Context) grom.DB {
  return m.client.WithContext(ctx)
}

func (m *mysql) ANY_OPERATION (ctx context Context) {
  // should only call m.client
  m.client(ctx).OP(...)
}
```

Reply:
Sure. I will update all the operations.

Reply:
Done !!